f3/granite.go

```diff
@@ -976,19 +977,123 @@ func (q *quorumState) HasWeakQuorumAgreement(cid TipSetID) bool {
 	return ok && cp.hasWeakQuorum
 }
 
-// Returns a list of the chains which have reached an agreeing strong quorum.
-// The order of returned values is not defined.
-func (q *quorumState) ListStrongQuorumAgreedValues() []ECChain {
-	var withQuorum []ECChain
+// Returns the value that has reached a strong quorum.
+// If more than one value fits this description, the function panics.
+func (q *quorumState) ListStrongQuorumAgreedValue() ECChain {
+	var withQuorum ECChain
 	for cid, cp := range q.chainSupport {
 		if cp.hasStrongQuorum {
-			withQuorum = append(withQuorum, q.chainSupport[cid].chain)
+			if withQuorum != nil {
+				panic(fmt.Sprintf("Found more than one value with a strong quorum: %v", withQuorum))
+			}
+			withQuorum = q.chainSupport[cid].chain
```

Comment:
It might be worth leaving a comment that avoiding an early return here is intentional so that we will detect the panic condition if it exists, because the panic is for safety.